### DISCUSSION TITLE
add a new slot

### DISCUSSION THREAD

**Discussion by rajasreemenon** [satijalab/seurat#5617] [Q&A]
Hi,
What is the best way to add a new slot (in addition to counts, data and scale.data)  to RNA assay . For example , a background corrected expression matrix.
Thanks,
Raji

**Comment by timoast** [comment]
If you want to add to the Seurat class definition you need to create a new class that inherits from Seurat. Here's an example: https://github.com/timoast/signac/blob/63d61bed2d6f43c7fd483b77eef0c84aa70952c9/R/objects.R#L98-L111

If you don't want to modify the class, you can store things in the `misc` slot

**Comment by rajasreemenon** [comment]
OK Thank you

**Comment by vivekruhela** [comment]
Hello, 

I am trying the same. Would you please give a simple example on PBMC public data? Thanks
